service.py

```diff
@@ -114,6 +114,42 @@ def createListItemFromFlatPlaylistItem(video):
 
     return listItem
 
+# get the index of the first video to be played in the submitted playlist url
+def playlistIndex(url, playlist):
+    if sys.version_info >= (3, 0):
```

Comment:
Yes it would be more pythonic to import everything at top anyway. 

On the other hand at least on my side importing bigger modules like e.g. yt_dlp or youtube_dl is way slower with the embedded kodi python interpreter (I just timed it using the log) than with the systems python shell. Maybe importing `urlparse, parse_qs` is super quick and it does not increase execution - otherwise I would try to not import them when we do not have a playlist and just a single video. 

Can you please check how fast the imports are on your machine (would be great to know for youtube_dl/yt_dlp as well), maybe its just something with my kodi instance.

Reply:
Changed to be as same as possible. Not really possible to unify them it turned out since L155 has a compound condition for importing these bigger modules anyway.